### ISSUE TITLE
Invalid module from IrToBitcodeKt.verifyModule

### ISSUE THREAD

**Issue by kpgalligan** [JetBrains/kotlin-native#1613]
I have a set of code that was auto-converted from Java that compiles but fails later in the build process. I've whittled that down to the following, which will fail to compile and run. You can replicate this in CLion with a basic "Hello World" project, and just copy the 'doubleException' method into it. The code never calls the method, but the build will fail.

```kotlin
fun main(args: Array<String>) {
    println("Hello, Native World!")
}

fun doubleException(value:Any):Int?{
    try{
        return (value as Long).toInt()
    }
    catch (e:ClassCastException) {
        if (value is CharSequence) {

            try {
                return value.toString().toInt()
            } catch (e2: NumberFormatException) {
                return null
            }
        }
        else{
            return null
        }
    }
}
```

The following, however, will pass. The only difference is moving the 'return null' to the end of the first catch block.

```kotlin
fun main(args: Array<String>) {
    println("Hello, Native World!")
}

fun doubleException(value:Any):Int?{
    try{
        return (value as Long).toInt()
    }
    catch (e:ClassCastException) {
        if (value is CharSequence) {

            try {
                return value.toString().toInt()
            } catch (e2: NumberFormatException) {

            }
        }

        return null
    }
}
```
The full build log is *long*, but the end and stacktrace are here:

```
error: compilation failed: Invalid module

 * Source files: hello.kt
 * Compiler version info: Konan: 0.7 / Kotlin: 1.2.50
 * Output kind: PROGRAM

exception: java.lang.Error: Invalid module
	at org.jetbrains.kotlin.backend.konan.llvm.IrToBitcodeKt.verifyModule(IrToBitcode.kt:147)
	at org.jetbrains.kotlin.backend.konan.llvm.IrToBitcodeKt.verifyModule$default(IrToBitcode.kt:138)
	at org.jetbrains.kotlin.backend.konan.Context.verifyBitCode(Context.kt:329)
	at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4.invoke(KonanDriver.kt:101)
	at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4.invoke(KonanDriver.kt)
	at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:140)
	at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:119)
	at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:34)
	at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:139)
	at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:90)
	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:68)
	at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:40)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:107)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:51)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:96)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:72)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:38)
	at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:171)
	at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:162)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:190)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:181)
	at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:34)
	at org.jetbrains.kotlin.backend.konan.util.UtilKt.profile(util.kt:29)
	at org.jetbrains.kotlin.cli.bc.K2Native$Companion.main(K2Native.kt:183)
	at org.jetbrains.kotlin.cli.bc.K2NativeKt.main(K2Native.kt:195)
	at org.jetbrains.kotlin.cli.utilities.MainKt.main(main.kt:27)
```

It appears to compile fine but fails later.

This test sample is using 0.7, but my original failing build is working from current master, so I assume there is no fix for this in later commits.

Obviously my original source isn't great and I'm just going to clean it up a bit, but I'm assuming the root cause of the error is something that might pop up at odd times and should be investigated.

**Comment by Albert-Gao** [comment]
@kpgalligan Hi, it's awesome. Could you please share how did you narrow it down? I have large code base where I have no idea where to start...

I inspected your code, seems I don't have this kind of logic in my code.

Thanks

**Comment by kpgalligan** [comment]
Usually when coding I'll periodically build everything and try it out, but in this particular case it was a lot of code being ported from Java, so I added a lot of kotlin without building it. At the end I ran into that issue. The easiest way to figure things out was to create a new branch in git, remove all of the new code, then start adding it back. I added a few classes and it built fine. Added a few more and it was fine. Added a few more and it failed. That left about 5 classes I needed to inspect, and I just narrowed it down from there.
